src/main/java/edu/cornell/kfs/module/cg/businessobject/lookup/CuAwardLookupableHelperServiceImpl.java

```diff
@@ -0,0 +1,137 @@
+package edu.cornell.kfs.module.cg.businessobject.lookup;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.kuali.kfs.kns.lookup.HtmlData;
+import org.kuali.kfs.kns.lookup.HtmlData.AnchorHtmlData;
+import org.kuali.kfs.kns.service.DocumentHelperService;
+import org.kuali.kfs.kns.web.struts.form.LookupForm;
+import org.kuali.kfs.kns.web.ui.Column;
+import org.kuali.kfs.kns.web.ui.ResultRow;
+import org.kuali.kfs.krad.util.GlobalVariables;
+import org.kuali.kfs.krad.util.KRADConstants;
+import org.kuali.kfs.module.ar.ArConstants;
+import org.kuali.kfs.module.cg.CGConstants;
+import org.kuali.kfs.module.cg.businessobject.lookup.AwardLookupableHelperServiceImpl;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.KFSPropertyConstants;
+import org.kuali.rice.kim.api.KimConstants;
+import org.kuali.rice.kim.api.permission.PermissionService;
+import org.kuali.rice.krad.bo.BusinessObject;
+
+import edu.cornell.kfs.sys.CUKFSConstants;
+
+@SuppressWarnings("deprecation")
+public class CuAwardLookupableHelperServiceImpl extends AwardLookupableHelperServiceImpl {
+    private static final long serialVersionUID = 1372707190594999024L;
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(CuAwardLookupableHelperServiceImpl.class);
+    
+    private static final String VIEW_INVOICES_LINK_VALUE = "View Invoices";
+    private static final String INVOICE_LINK_COLUMN_NAME = "invoiceLink";
+    
+    protected DocumentHelperService documentHelperService;
+    protected PermissionService permissionService;
+    
+    @Override
+    public List<HtmlData> getCustomActionUrls(BusinessObject businessObject, List pkNames) {
+        LOG.debug("getCustomActionUrls, entering");
+        List<HtmlData> anchorHtmlDataList = new ArrayList<HtmlData>();
+        
+        if (canInitAward() && allowsMaintenanceEditAction(businessObject)) {
+            anchorHtmlDataList.add(getUrlData(businessObject, KRADConstants.MAINTENANCE_EDIT_METHOD_TO_CALL, pkNames));
+            if (allowsMaintenanceNewOrCopyAction()) {
```

Comment:
Because of how the "if" blocks are being nested, this current setup will only allow the "copy" action to those who have both the edit and new/copy permissions. Are we expecting our permission setup to either allow both or none? If so, then I am okay with moving this forward, and we can clean this up in a future story if necessary.

Reply:
I double checked with Kris, and the way CU handles awards, we wouldn't have the case of copying without being able edit.  So I think this setup is fine for now.

Reply:
Ok, sounds good. Thanks for checking!